Framework/Algorithms/test/ResampleXTest.h

```diff
@@ -161,6 +170,258 @@ class ResampleXTest : public CxxTest::TestSuite {
+    do_testResampleXEventWorkspace(WEIGHTED_NOTIME, false, true, true);
+  }
+
+  // Set-up a generic function for running tests for Workspace2D
+  void do_testResampleXWorkspace2D(bool inPlace, bool withDistribution) {
+
+    std::string inName("test_inEvent");
+    std::string outName("test_inEvent_output");
+    if (inPlace)
+      outName = inName;
+
+    Workspace2D_sptr test_in2D = Create2DWorkspace(100, 2);
+    test_in2D->setDistribution(withDistribution);
+    AnalysisDataService::Instance().add(inName, test_in2D);
+
+    ResampleX alg;
+    alg.initialize();
+    alg.setPropertyValue("InputWorkspace", inName);
+    alg.setPropertyValue("OutputWorkspace", outName);
+    alg.setPropertyValue("Xmin", "0.0,0.0");
+    alg.setPropertyValue("Xmax", "100,150");
+    alg.setPropertyValue("NumberBins", "50");
+
+    TS_ASSERT(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+
+    MatrixWorkspace_sptr outWS;
+    EventWorkspace_sptr eventOutWS;
+    TS_ASSERT_THROWS_NOTHING(
+        outWS = boost::dynamic_pointer_cast<MatrixWorkspace>(
+            AnalysisDataService::Instance().retrieve(outName)));
+    TS_ASSERT(outWS);
+    if (!outWS)
+      return;
+
+    // Define tolerance for ASSERT_DELTA
+    double tolerance = 1.0e-10;
+
+    // Check the first workspace index (from 0 to 100)
+    auto &X1 = outWS->x(0);
+    auto &Y1 = outWS->y(0);
+    auto &E1 = outWS->e(0);
+
+    TS_ASSERT_EQUALS(X1.size(), 51);
+    TS_ASSERT_DELTA(X1[0], 0.0, tolerance);
+    TS_ASSERT_DELTA(X1[1], 2.0, tolerance);
+    TS_ASSERT_DELTA(X1[2], 4.0, tolerance);
+
+    TS_ASSERT_EQUALS(Y1.size(), 50);
+    TS_ASSERT_DELTA(Y1[0], 6.0, tolerance);
+    TS_ASSERT_DELTA(Y1[1], 8.0, tolerance);
+    TS_ASSERT_DELTA(Y1[2], 8.0, tolerance);
+
+    TS_ASSERT_EQUALS(E1.size(), 50);
+    TS_ASSERT_DELTA(E1[0], sqrt(6.0), tolerance);
+    TS_ASSERT_DELTA(E1[1], sqrt(8.0), tolerance);
+
+    // Check the second workspace index (from 0 to 150)
+    auto &X2 = outWS->x(1);
+    auto &Y2 = outWS->y(1);
+    auto &E2 = outWS->e(1);
+
+    TS_ASSERT_EQUALS(X2.size(), 51);
+    TS_ASSERT_DELTA(X2[0], 0.0, tolerance);
+    TS_ASSERT_DELTA(X2[1], 3.0, tolerance);
+    TS_ASSERT_DELTA(X2[2], 6.0, tolerance);
+
+    TS_ASSERT_EQUALS(Y2.size(), 50);
+    TS_ASSERT_DELTA(Y2[0], 10.0, tolerance);
+    TS_ASSERT_DELTA(Y2[1], 12.0, tolerance);
+    TS_ASSERT_DELTA(Y2[2], 12.0, tolerance);
+
+    TS_ASSERT_EQUALS(E2.size(), 50);
+    TS_ASSERT_DELTA(E2[0], sqrt(10.0), tolerance);
+    TS_ASSERT_DELTA(E2[1], sqrt(12.0), tolerance);
+
+    AnalysisDataService::Instance().remove(inName);
+    AnalysisDataService::Instance().remove(outName);
+  }
+
+  void testWorkspace2D_InPlace_NoDistribution() {
+    do_testResampleXWorkspace2D(true, false);
+  }
+
+  void testWorkspace2D_NotInPlace_NoDistribution() {
+    do_testResampleXWorkspace2D(false, false);
+  }
+
+  void xtestWorkspace2D_InPlace_WithDistribution() {
+    do_testResampleXWorkspace2D(true, true);
```

Comment:
If you disable a test, you should definitely put a comment explaining why it is disabled, referencing the issue you created.